examples/python/debug/get_stats.py

```diff
@@ -20,8 +20,13 @@
 import logging
 import argparse
 import grpc
-from grpc_channelz.v1 import channelz_pb2
-from grpc_channelz.v1 import channelz_pb2_grpc
+
+try:
+    from src.python.grpcio_channelz.grpc_channelz.v1 import channelz_pb2
+    from src.python.grpcio_channelz.grpc_channelz.v1 import channelz_pb2_grpc
+except ImportError:
+    from grpc_channelz.v1 import channelz_pb2
+    from grpc_channelz.v1 import channelz_pb2_grpc
```

Comment:
I'm afraid this isn't quite the same problem. The generated code references `src.python.grpcio_channelz.grpc_channelz.v1.channelz_pb2`, whereas user-written code references `grpc_channelz.v1`. While both of these import paths refer to the same file, they end up instantiating two separate module objects with incompatible classes.

This *could* be solved by https://github.com/grpc/grpc/pull/18808, but in the general case, it would require quite a bit of plumbing and probably an addition to our `py_proto_library` and `py_grpc_library` APIs. Internally, we need to strip `google3.third_party.py.`. As a rule available for general use in OSS, there could be an arbitrary collection of prefixes you'd like to strip. In the case of just our repo, we'd probably need to strip
 - `src.python.grpcio_channelz.`
 - `src.python.grpcio_health_checking.`
 - `src.python.grpcio_reflection.`

The same issue will come up every time someone wants to manage code with Bazel and `setuptools` without code changes. That's essentially the problem we've seen internally, anyway.

Since this is only currently possible by modifying the C++ code for the plugin, in order to make it extensible, I think we'd need to:
 - Add a new flag to the `protoc` plugin making this option accessible to users without recompiling.
 - Add an option to our bazel macro to make use of the new flag.

But this is a non-negligible chunk of work that can be done in a backwards-compatible manner. For the moment, I'd rather add a TODO and pay this down in a future PR.

Reply:
I don't have a better solution than your proposal. A TODO looks good to me.